lib/ansible/plugins/action/reboot.py

```diff
@@ -103,9 +103,14 @@ def get_system_boot_time(self):
         if '1970-01-01 00:00' in command_result['stdout']:
             command_result = self._low_level_execute_command('uptime -s', sudoable=self.DEFAULT_SUDOABLE)
 
+        # This is a last resort for bare Linux systems (e.g. OpenELEC) where 'who -b' or 'uptime -s' are not supported.
+        # Other options like parsing /proc/uptime or default uptime output are less reliable than this
+        if command_result['rc'] != 0:
+            command_result = self._low_level_execute_command('cat /proc/sys/kernel/random/boot_id', sudoable=self.DEFAULT_SUDOABLE)
+
         if command_result['rc'] != 0:
             raise AnsibleError("%s: failed to get host boot time info, rc: %d, stdout: %s, stderr: %s"
-                               % (self._task.action, command_result.rc, to_native(command_result['stdout']), to_native(command_result['stderr'])))
+                               % (self._task.action, command_result['rc'], to_native(command_result['stdout']), to_native(command_result['stderr'])))
```

Comment:
I would aggregate both errors here, both 'uptime' and 'cat /proc..'

Reply:
Agree with this. Don't want to swallow the output from the initial command.

Long term, it feels like we may need to change distribution to being an attribut, that way we can run `cat ...` for Linux hosts, and use `who -b` for others, removing the need for `uptime -s` and a hard coded tree of commands to attempt.

Reply:
I fixed it so we collect all the stdout and stderr information in case we need it for the AnsibleError.

The problem with having only a single guaranteed command per OS is
1. You would need to know the platform/OS you're working with
2. I don't think it's that clear-cut

So this fallback method, while not necessarily 100% clean, it does the job effectively.
And `who -b` and `uptime -s` are very sensible things to do before resorting to the random boot_id. (Which may not be present in all Linux kernels either ?)

Reply:
My next question was going to be if the random boot ID is widely available. I guess I'll have to do some research to see when it was added.

I created a WIP PR (#45656) exploring what it would look like to add platform and dist to the plugin. I'm not terribly happy with it, but it's a start (and it accounts for Alpine, which is particularly annoying since it lacks both `who` and `uptime` for no good reason (IMO)).

That PR also changes to using the random boot ID by default rather than `who -b`. That has the nice side effect of being able to remove the `uptime -s` command for systems that lack a RTC, but at the cost of all the added complexity of platfrom and dist checking.